framework/src/application/node/transport/transport.ts

```diff
@@ -0,0 +1,524 @@
+		return this._transactionPoolModule;
+	}
+
+	public get chainModule(): Chain {
+		return this._chainModule;
+	}
+
+	public get processorModule(): Processor {
+		return this._processorModule;
+	}
+
+	public get broadcaster(): Broadcaster {
+		return this._broadcaster;
+	}
+
+	public handleBroadcastTransaction(transaction: BaseTransaction): void {
+		this._broadcaster.enqueueTransactionId(transaction.id);
+		this._channel.publish('app:transaction:new', transaction.toJSON());
+	}
+
+	public async handleBroadcastBlock(blockJSON: BlockJSON): Promise<unknown> {
+		if (this._synchronizerModule.isActive) {
+			this._logger.debug(
+				'Transport->onBroadcastBlock: Aborted - blockchain synchronization in progress',
+			);
+			return null;
+		}
+		return this._channel.publishToNetwork('sendToNetwork', {
+			event: 'postBlock',
+			data: {
+				block: blockJSON,
+			},
+		});
+	}
+
+	public async handleRPCGetBlocksFromId(
+		data: RPCBlocksByIdData,
+		peerId: string,
+	): Promise<BlockJSON[]> {
+		const errors = validator.validate(schemas.getBlocksFromIdRequest, data);
+
+		if (errors.length) {
+			const error = `${errors[0].message}`;
+
+			this._logger.warn(
+				{
+					err: error,
+					req: data,
+				},
+				'getBlocksFromID request validation failed',
+			);
+			await this._channel.invoke('app:applyPenaltyOnPeer', {
+				peerId,
+				penalty: 100,
+			});
+			throw new Error(error);
+		}
+
+		// Get height of block with supplied ID
+		const lastBlock = await this._chainModule.dataAccess.getBlockHeaderByID(
+			data.blockId,
+		);
+		if (!lastBlock) {
+			throw new Error(`Invalid blockId requested: ${data.blockId}`);
+		}
+
+		const lastBlockHeight = lastBlock.height;
+
+		// Calculate max block height for database query
+		// 15kb * 103 is about 1.5MB where it's half of 3MB payload limit
+		const fetchUntilHeight = lastBlockHeight + 103;
+
+		const blocks = await this._chainModule.dataAccess.getBlocksByHeightBetween(
+			lastBlockHeight + 1,
+			fetchUntilHeight,
+		);
+
+		return blocks && blocks.map(block => this._chainModule.serialize(block));
+	}
+
+	public async handleRPCGetGetHighestCommonBlock(
+		data: RPCHighestCommonBlockData,
+		peerId: string,
+	): Promise<BlockHeaderJSON | null> {
+		const valid = validator.validate(
+			schemas.getHighestCommonBlockRequest,
+			data,
+		);
+
+		if (valid.length) {
+			const err = valid;
+			const error = `${err[0].message}: ${err[0].dataPath}`;
+			this._logger.warn(
+				{
+					err: error,
+					req: data,
+				},
+				'getHighestCommonBlock request validation failed',
+			);
+			await this._channel.invoke('app:applyPenaltyOnPeer', {
+				peerId,
+				penalty: 100,
+			});
+			throw new Error(error);
+		}
+
+		const commonBlock = await this._chainModule.getHighestCommonBlock(data.ids);
+
+		return commonBlock
+			? this._chainModule.serializeBlockHeader(commonBlock)
+			: null;
+	}
+
+	public async handleEventPostBlock(
+		data: EventPostBlockData,
```

Comment:
I can be done in other issue, but if it comes from outside of framework, probably it's better to treat it as `unknown` at the beginning, and after it's validated, cast to the actual type?